source/_docs/composer.md

```diff
@@ -1,175 +1,87 @@
 ---
-title: Managing Custom Code with Composer
-description: Understand how to manage custom plugins and themes for WordPress sites using Composer.
-tags: [workflow]
-categories: []
+title: Composer Fundamentals and Workflows
+description: Start with Composer basics then explore suggested workflows for WordPress and Drupal sites on Pantheon.
+tags: [automation, workflow]
 ---
-Extending WordPress or Drupal with custom code is a common part of the development lifecycle. In order to optimize workflows, it's essential that custom plugins, modules, and themes are maintained independently from your sites and projects so they're easily reused.
+Composer is a PHP dependency manager that provides an alternative, more modern way to manage the external code used by a WordPress or Drupal site. At it's primary level, Composer needs:
+
+- A list of dependencies
+- A place to put the dependencies
+
+Understanding how Composer can be used independent of Drupal or WordPress is a good place to learn more about the general concepts. For a summary of basic usage, see [Composer's own documentation](https://getcomposer.org/doc/01-basic-usage.md).
+
+## Dependencies and Packages
+Composer encourages a mental model where code not written specifically for a given project is considered a "dependency". Dependencies for WordPress and Drupal include core, plugins, contrib modules, themes, and libraries. A single dependency, such as a theme, is referred to as a "package".
+
+By default, Composer can only see packages listed on [The PHP Package Repository](https://packagist.org/) which do not include Drupal or WordPress packages. Additional repositories must be configured for Composer to use packages not found in the default repository. Each framework provides it's own respective package repository so dependencies can be managed with Composer:
+
+- WordPress: [https://wpackagist.org](https://wpackagist.org)
+- Drupal 8: [https://packages.drupal.org/8](https://packages.drupal.org/8)
+- Drupal 7: [https://packages.drupal.org/7](https://packages.drupal.org/7)
+
+Site's created from Pantheon's example repositories already include the appropriate package repository within the `composer.json` file.
+
+## Nested Docroot
```

Comment:
This section feels a little wierd to me. Composer can be used with a site with no nested docroot, and some of the arguments for nested docroots have nothing to do with composer (keeping project config out of the site root, including developer docs and build files outside of the web-accessible directories, etc). The push for both concepts kind of end up coming from the same people and started around the same timeframe, but they are actually separate. Its not a big deal, but I feel we might be over simplifying this and suggesting the two ideas as interrelated when they sort of aren't. I don't feel super strongly about this, what do you think @greg-1-anderson  and @stevector ?

Reply:
Right now, Pantheon maintains no upstreams with nested docroots. So, for most users, I would say that if you use a nested docroot, you should manage your site updates with Composer. If we think folks want nested docroots without Composer, then we should start maintaining some upstreams for them to use.

On the flip side, while it is technically possible to use Composer without using a nested docroot, doing this will often get the user into a situation where there are a lot of conflicts that need to be resolved. It seems much safer to me to just get everyone who wants a Composer-managed site to use a nested docroot.

We could say "SHOULD be installed in a subdirectory" instead of MUST, if you prefer, though.